tests/integration/pilot/revision_view_test.go

```diff
@@ -0,0 +1,157 @@
+			defaultRevNs := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "istioctl-rev-default",
+				Inject: true,
+			})
+			canaryRevNs := namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix:   "istioctl-rev-canary",
+				Inject:   false,
+				Revision: "canary",
+			})
+			builder := echoboot.NewBuilder(ctx)
+
+			var defaultEchoInstance echo.Instance
+			var canaryEchoInstance echo.Instance
+			builder.With(&defaultEchoInstance, echo.Config{Namespace: defaultRevNs}).
+				With(&canaryEchoInstance, echo.Config{Namespace: canaryRevNs}).
+				BuildOrFail(ctx)
+			testRevisionListing(ctx, istioCtl, defaultRevNs, canaryRevNs)
+		})
+}
+
+func testRevisionListing(ctx framework.TestContext, istioCtl istioctl.Instance,
+	defaultRevNs namespace.Instance, canaryRevNs namespace.Instance) {
+	listCmd := []string{
+		"experimental", "revision", "list",
+		"-d", operator.ManifestPath,
+		"-o", "json",
+	}
+	jsonOut, _, err := istioCtl.InvokeOrFail(ctx, listCmd)
+	if err != nil {
+		ctx.Fatalf("revision listing should have been successful, "+
+			"but got an error instead: %v", err)
+	}
+	var descriptions map[string]*cmd.RevisionDescription
+	err = json.Unmarshal([]byte(jsonOut), descriptions)
+	if err != nil {
+		ctx.Fatalf("error while unmarshaling revision list output: %v", err)
+	}
+	expectedRevisions := []string{"default", "revision"}
+	for rev, rd := range descriptions {
+		found := false
+		for _, r := range expectedRevisions {
+			if r == rev {
+				found = true
+				break
+			}
+		}
+		if !found {
+			ctx.Fatalf("unexpected revision found: %s", rev)
+		}
+		if len(rd.IstioOperatorCRs) != 1 {
+			ctx.Fatalf("expected a single IstioOperator CR for %s, found %d",
+				rev, len(rd.IstioOperatorCRs))
+		}
+		ctx.Logf("installed IstioOperator CR (profile=%s) components: %v\n",
+			rd.IstioOperatorCRs[0].Profile, rd.IstioOperatorCRs[0].Components)
+		if len(rd.ControlPlanePods) == 0 {
+			ctx.Fatalf("no control plane pods listed for revision %s", rev)
+		}
+		if len(rd.Webhooks) == 0 {
+			ctx.Fatalf("no mutating webhook listed for revision %s", rev)
+		}
+	}
+}
+
+func TestInvalidFormat(t *testing.T) {
+	framework.
+		NewTest(t).
+		RequiresSingleCluster().
+		Features("installation.istioctl.revision_centric_view").
+		Run(func(ctx framework.TestContext) {
+			type invalidFormatTest struct {
+				name    string
+				command []string
+			}
+			istioCtl := istioctl.NewOrFail(ctx, ctx, istioctl.Config{})
+			for _, t := range []invalidFormatTest{
+				{name: "list", command: []string{"x", "revision", "list", "-o", "mystery"}},
+				{name: "describe", command: []string{"x", "revision", "describe", "-o", "mystery"}},
+			} {
+				ctx.NewSubTest(t.name).Run(func(sctx framework.TestContext) {
+					_, _, fErr := istioCtl.Invoke(listCmd)
+					if fErr == nil {
+						ctx.Fatalf("expected error due to invalid format, but got nil")
+					}
+				})
+			}
+		})
+}
+
+func TestValidRevisionDescription(t *testing.T) {
```

Comment:
missing test?

Reply:
Yes. That is yet to be written.

Reply:
Wrote a couple of integration tests for both list and describe commands. Lets see if any more tests are required